sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ParserUtilsSuite.scala

```diff
@@ -61,5 +102,87 @@ class ParserUtilsSuite extends SparkFunSuite {
+    assert(command(descFuncContext) == "describe function extended bar")
+    assert(command(showDbsContext) == "show databases like 'identifier_with_wildcards'")
+  }
+
+  test("operationNotAllowed") {
+    val errorMessage = "parse.fail.operation.not.allowed.error.message"
+    val e = intercept[ParseException] {
+      operationNotAllowed(errorMessage, showFuncContext)
+    }.getMessage
+    assert(e.contains("Operation not allowed"))
+    assert(e.contains(errorMessage))
+  }
+
+  test("checkDuplicateKeys") {
+    val properties = Seq(("a", "a"), ("b", "b"), ("c", "c"))
+    checkDuplicateKeys[String](properties, createDbContext)
+
+    val properties2 = Seq(("a", "a"), ("b", "b"), ("a", "c"))
+    val e = intercept[ParseException] {
+      checkDuplicateKeys(properties2, createDbContext)
+    }.getMessage
+    assert(e.contains("Found duplicate keys"))
+  }
+
+  test("source") {
+    assert(source(setConfContext) == "set example.setting.name=setting.value")
+    assert(source(showFuncContext) == "show functions foo.bar")
+    assert(source(descFuncContext) == "describe function extended bar")
+    assert(source(showDbsContext) == "show databases like 'identifier_with_wildcards'")
+  }
+
+  test("remainder") {
+    assert(remainder(setConfContext) == "")
+    assert(remainder(showFuncContext) == "")
+    assert(remainder(descFuncContext) == "")
+    assert(remainder(showDbsContext) == "")
+
+    assert(remainder(setConfContext.SET.getSymbol) == " example.setting.name=setting.value")
+    assert(remainder(showFuncContext.FUNCTIONS.getSymbol) == " foo.bar")
+    assert(remainder(descFuncContext.EXTENDED.getSymbol) == " bar")
+    assert(remainder(showDbsContext.LIKE.getSymbol) == " 'identifier_with_wildcards'")
+  }
+
+  test("string") {
+    assert(string(showDbsContext.pattern) == "identifier_with_wildcards")
+    assert(string(createDbContext.comment) == "database_comment")
+
+    assert(string(createDbContext.locationSpec.STRING) == "/home/user/db")
+  }
+
+  test("position") {
+    assert(position(setConfContext.start) == Origin(Some(1), Some(0)))
+    assert(position(showFuncContext.stop) == Origin(Some(1), Some(19)))
+    assert(position(descFuncContext.describeFuncName.start) == Origin(Some(1), Some(27)))
+    assert(position(createDbContext.locationSpec.start) == Origin(Some(3), Some(27)))
+    assert(position(emptyContext.stop) == Origin(None, None))
+  }
+
+  test("assert") {
+    val f1 = { ctx: ParserRuleContext =>
+      ctx.children != null && !ctx.children.isEmpty
+    }
+    val message = "ParserRuleContext should not be empty."
+    ParserUtils.assert(f1(showFuncContext), message, showFuncContext)
+
+    val e = intercept[ParseException] {
+      ParserUtils.assert(f1(emptyContext), message, emptyContext)
+    }.getMessage
+    assert(e.contains(message))
+  }
+
+  test("withOrigin") {
+    val ctx = createDbContext.locationSpec
+    val current = CurrentOrigin.get
+    val location = withOrigin(ctx) {
```

Comment:
you also need to check if the origin of location to make this a complete tests, e.g.:
```scala
val (location, origin) = withOrigin(ctx) {
  (string(ctx.string), CurrentOrigin.get)
}
assert(origin, Origin(Some(2), Some(37)))
```